### PR TITLE
Add column comments to Snowflake views

### LINKED ISSUE #17: Snowflake Persist Docs (Materialization == Views, column level descriptions)
### Describe the feature
Given the database adapter is Snowflake, materialization type is View and persist docs is set true - create the view using the following snowflake syntax so that column descriptions from dbt schema.yml files are include:

CREATE OR REPLACE VIEW <VIEWNAME> (COL1 COMMENT 'COMMENT', COLN COMMENT 'COMMENT') AS SELECT COL1, COLN FROM TABLE;

### Describe alternatives you've considered
I am looking into submitting a PR myself but would benefit from oversight. We have also considered contacting Snowflake to include alter capabilities for columns within a view but believe this to be another alternative.

### Additional context
Snowflake 

### Who will this benefit?
All users of Snowflake who heavily rely on views and would like to use persist docs to capture doc details within the database itself for external dependencies

### Are you interested in contributing this feature?
Yes - more than happy to help - just require a push in the right direction

### PR DESCRIPTION
**Pull request by spencer-taylor-workrise**
resolves #17 

### Description

Adds functionality to persist column descriptions to Snowflake views upon creation.
Expands column name testing (upper/lowercase) for the column-level description integration tests.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-snowflake next" section.

### CODE REVIEW

**Comment by jtcohen6** [comment]
Closing and re-opening to trigger adapter integration tests, now that I've added `ok to test` label

File: dbt/include/snowflake/macros/adapters.sql [hunk 1]
```diff
@@ -42,11 +42,27 @@
   {%- set secure = config.get('secure', default=false) -%}
   {%- set copy_grants = config.get('copy_grants', default=false) -%}
   {%- set sql_header = config.get('sql_header', none) -%}
+  {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
+  {% set model_columns = model.columns %}
 
+  {% set query_columns = get_columns_in_query(sql + ' limit 0') %}
```
Comment (jtcohen6) [inline dbt/include/snowflake/macros/adapters.sql:48]: 1. Let's avoid running this query if not `config.persist_column_docs()`, most likely by nesting it within the existing conditional below
2. The [`get_columns_in_query` macro](https://github.com/dbt-labs/dbt-core/blob/d5d16f01f4d18619ec4200728860dc64f5ce70c9/core/dbt/include/global_project/macros/adapters/columns.sql#L24) already handles adding `where false limit 0` + wrapping in a subquery
```suggestion
  {% set query_columns = get_columns_in_query(sql) %}
```

File: dbt/include/snowflake/macros/adapters.sql [hunk 2]
```diff
@@ -42,11 +42,27 @@
...
+  {% if config.persist_column_docs() -%}
+    (
+      {% for column_name in query_columns %}
+        {% if (column_name|upper in model_columns) or (column_name in model_columns) %}
+          {{ adapter.quote(column_name) if model_columns[column_name]['quote'] else column_name }} COMMENT '{{ model_columns[column_name]['description']}}'
+        {% else %}
+          {{column_name}}
+        {% endif %}
+        {{ ", " if not loop.last else "" }}
+      {% endfor %}
```
Comment (jtcohen6) [inline dbt/include/snowflake/macros/adapters.sql:62]: Nice work sketching this out! I think we should have two goals:
- avoid duplicating logic between here + `snowflake__alter_column_comment` down below
- keep the `create_view_as` macro clean + legible

I've got some pseudo-code below with ideas of how to do it. Feel free to take/leave/adapt as you see fit.

First, we can wrap the common bits of comment generation in a new macro:
```sql
{% macro get_column_comment_sql(column_name, column_dict) %}
    {{ adapter.quote(column_name) if column_dict[column_name]['quote'] else column_name }} COMMENT $${{ column_dict[column_name]['description'] | replace('$', '[$]') }}$$
{% endmacro %}
```

Then, `create_view_as` can call a separate macro if we're persisting column-level docs:
```sql
{% if config.persist_column_docs() -%}
    {% set model_columns = model.columns %}
    {% set query_columns = get_columns_in_query(sql) %}
    {{ get_persist_docs_column_list(model_columns, query_columns) }}
{% endif %} 
```

That macro would look like:

```sql
{% macro get_persist_docs_column_list(model_columns, query_columns) %}
    {% for column_name in query_columns %}
        {% if (column_name|upper in model_columns) or (column_name in model_columns) %}
          {{ get_column_comment_sql(column_name, model_columns) }}  
        {% else %}
          {{ column_name }}
        {% endif %}
        {{ ", " if not loop.last else "" }}
      {% endfor %}
{% endmacro %}
```
And then the other macro down below:
```sql
{% macro snowflake__alter_column_comment(relation, column_dict) -%}
    {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
    alter {{ relation.type }} {{ relation }} alter
    {% for column_name in column_dict if (column_name in existing_columns) or (column_name|upper in existing_columns) %}
        {{ get_column_comment_sql(column_name, column_dict) }} {{ ',' if not loop.last else ';' }}
    {% endfor %}
{% endmacro %}
```

File: dbt/include/snowflake/macros/adapters.sql [hunk 3]
```diff
@@ -42,11 +42,27 @@
   {%- set secure = config.get('secure', default=false) -%}
   {%- set copy_grants = config.get('copy_grants', default=false) -%}
   {%- set sql_header = config.get('sql_header', none) -%}
+  {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
```
Comment (jtcohen6) [inline dbt/include/snowflake/macros/adapters.sql:45]: Do we need to get `existing_columns`?

**Review by jtcohen6 — COMMENTED** [review]
Nice work so far on this @spencer-taylor-workrise! Thanks for proving out the approach. A few comments from me, thinking through best ways to keep the logic organized

File: dbt/include/snowflake/macros/adapters.sql (hunk 3 above)
Comment (spencer-taylor-workrise) [reply dbt/include/snowflake/macros/adapters.sql:45]: Nope, I think this is left over from a previous version I was testing out. Removing this shortly.

File: dbt/include/snowflake/macros/adapters.sql (hunk 1 above)
Comment (spencer-taylor-workrise) [reply dbt/include/snowflake/macros/adapters.sql]: Updated in 66312f7.

File: dbt/include/snowflake/macros/adapters.sql (hunk 2 above)
Comment (spencer-taylor-workrise) [reply dbt/include/snowflake/macros/adapters.sql]: Updated in 66312f7.

**Comment by spencer-taylor-workrise** [comment]
Hey @jtcohen6 ! Thanks for the comments -- completely agree with the logic there. These have all been resolved in the most recent commit.

**Review by jtcohen6 — APPROVED** [review]
@spencer-taylor-workrise Nice work, on a quick turnaround! This looks good to me :)

**Pull request merged**